pkg/manifests/common.go

```diff
@@ -15,6 +15,34 @@ func GetTopLevelLabels() map[string]string { // this is a function to avoid any
 	return map[string]string{"app.kubernetes.io/managed-by": operatorName}
 }
 
+// Checks the first set of labels has the labels of the other passed in sets
+func HasRequiredLabels(checkedLabels map[string]string, labelSet ...map[string]string) bool {
+	if len(checkedLabels) == 0 {
+		return false
+	}
+
+	labelSetLen := 0
+	for _, set := range labelSet {
+		labelSetLen += len(set)
+	}
+	if len(checkedLabels) < labelSetLen {
```

Comment:
how does using length verification work if two labelSets have the same label we are checking, then it would have a longer length. If we aren't deduping the desired labels then length equality has no guarantee